src/main/java/uk/gov/hmcts/reform/divorce/orchestration/service/impl/CaseServiceImpl.java

```diff
@@ -55,13 +56,16 @@
     private String caseType;
 
     @Override
-    public Map<String, Object> submitDraftCase(final Map<String, Object> caseData, final String authToken) {
-
-        final Map<String, Object> payload = cmsClient.submitDraftCase(caseData, authToken);
-
-        log.info("Case with CASE ID: {} submitted", payload.get(ID));
-
-        return payload;
+    public Map<String, Object> submitDraftCase(final Map<String, Object> caseData, final String authToken) throws CaseAlreadyExistsException {
+        try {
+            final GetCaseResponse getCaseResponse = getCase(authToken);
```

Comment:
Instead of using the getCase method, it might be better to pullout parts of the getCase method into a private method then the control of flow does not rely on an exception.  The methods below could become a seperate private method:

User user = getUser(authorizationToken);
List<CaseDetails> caseDetailsList = getCaseListForUser(user)
return filterOutAmendedCases(caseDetailsList);

Reply:
Shaun's suggestion would be better to implement